charmcraft.yaml

```diff
@@ -0,0 +1,82 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: loki-worker-k8s
+type: charm
+assumes:
+  - k8s-api
+  - juju >= 3.5.0
+
+summary: Loki for Kubernetes clusters.
+description: |
+  Grafana Loki is an open source software project that provides a scalable long-term storage for Prometheus.
+  This charm deploys and operates Loki on Kubernetes clusters
+
+# TODO: Write documentation
+# docs: https://discourse.charmhub.io/t/mimir-worker-k8s-index/13464
+# source: https://github.com/canonical/mimir-worker-k8s-operator
+# issues: https://github.com/canonical/mimir-worker-k8s-operator/issues
+
+bases:
+  - build-on:
+    - name: ubuntu
+      channel: "22.04"
+    run-on:
+    - name: ubuntu
+      channel: "22.04"
+
+parts:
+  charm:
+    charm-binary-python-packages:
+      - "pydantic>=2"
+      - "cryptography"
+      - "jsonschema"
+
+requires:
+  loki-cluster:
+    interface: loki_cluster
+    description: |
+      The requested set of roles for of the worker (this charm) is requested by
+      the coordinator over relation data.
```

Comment:
I think my change suggestion here was wrong: the roles are determined by worker's config options.